core/e2e/tests/blockstore_server.rs

```diff
@@ -29,6 +29,21 @@ fn create_content() -> Vec<u8> {
         .collect()
 }
 
+async fn wait_for_origin_propagation(swarm: &Swarm, node: &NodePublicKey, hash: &[u8; 32]) {
+    let query_runner = swarm.get_query_runner(node).unwrap();
+    let mut tries = 0;
+    loop {
```

Comment:
we could use poll_until here but I'm okay with this as is. But if you want to make the switch here's an example:
https://github.com/fleek-network/lightning/blob/b08752c7f3779fabe9036f8392e22b9fbe51d384/core/e2e/tests/eth.rs#L72-L88

Reply:
Nice!! Applied it.